addons/l10n_pe_pos/models/res_partner.py

```diff
@@ -0,0 +1,13 @@
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    @api.ondelete(at_uninstall=False)
+    def _pe_unlink_except_master_data(self):
+        consumidor_final_anonimo = self.env.ref("l10n_pe_pos.partner_pe_cf")
+        if consumidor_final_anonimo in self:
+            raise UserError(_("Deleting this partner is not allowed."))
```

Comment:
"Deleting the partner %s is not allowed because it's required by..."

Take into account we could remove several partners at once.